File: src/main/java/tn/esprit/models/Destination.java
package tn.esprit.models;
import java.util.Arrays;
import java.util.Objects;

public class Destination {

    private int id;
    private String name;
    private String img; // Changed from byte[] to String
    private String state;
    private String address;
    private float price; // Changed from prix to price
    private String description;

    public Destination() {
    }

    public Destination(String name, String img, String state, String address, float price, String description) { // Changed img to String
        this.name = name;
        this.img = img;
        this.state = state;
        this.address = address;
        this.price = price; // Changed prix to price
        this.description = description;
    }

    public Destination(int id, String name, String img, String state, String address, float price, String description) { // Changed img to String
        this.id = id;
        this.name = name;
        this.img = img;
        this.state = state;
        this.address = address;
        this.price = price; // Changed prix to price
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImg() { // Changed return type to String
        return img;
    }

    public String getState() {
        return state;
    }

    public String getAddress() {
        return address;
    }

    public float getPrice() { // Changed getPrix to getPrice
        return price;
    }

    public String getDescription() {
        return description;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setImg(String img) { // Changed parameter type to String
        this.img = img;
    }

    public void setState(String state) {
        this.state = state;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setPrice(float price) { // Changed setPrix to setPrice
        this.price = price;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Destination{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", img='" + img + '\'' +
                ", state='" + state + '\'' +
                ", address='" + address + '\'' +
                ", price=" + price +
                ", description='" + description + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        Destination that = (Destination) obj;
        return id == that.id &&
                Float.compare(that.price, price) == 0 &&
                name.equals(that.name) &&
                Objects.equals(img, that.img) && // Changed from img.equals(that.img)
                state.equals(that.state) &&
                address.equals(that.address) &&
                description.equals(that.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, img, state, address, price, description);
    }
}

File: src/Entity/Destination.php
<?php

namespace App\Entity;

use App\Repository\DestinationRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: DestinationRepository::class)]
class Destination
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;


    
    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le nom ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "Le nom ne peut pas dépasser {{ limit }} caractères.")]
    private ?string $name = null;

    #[ORM\Column(length: 255)]

    private ?string $img = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'état ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "L'état ne peut pas dépasser {{ limit }} caractères.")]
    private ?string $state = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'adresse ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "L'adresse ne peut pas dépasser {{ limit }} caractères.")]
    private ?string $address = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le prix ne peut pas être vide.")]
    #[Assert\Type(type: "float", message: "Le prix doit être un nombre.")]
    private ?float $prix = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La description ne peut pas être vide.")]
    private ?string $description = null;

    #[ORM\OneToMany(targetEntity: Reservation::class, mappedBy: 'Destination')]
    private Collection $reservation;

    public function __construct()
    {
        $this->reservation = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): static
    {
        $this->img = $img;

        return $this;
    }

    public function getState(): ?string
    {
        return $this->state;
    }

    public function setState(string $state): static
    {
        $this->state = $state;

        return $this;
    }

    public function getAddress(): ?string
    {
        return $this->address;
    }

    public function setAddress(string $address): static
    {
        $this->address = $address;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    /**
     * @return Collection<int, Reservation>
     */
    public function getReservation(): Collection
    {
        return $this->reservation;
    }

    public function addReservation(Reservation $reservation): static
    {
        if (!$this->reservation->contains($reservation)) {
            $this->reservation->add($reservation);
            $reservation->setDestination($this);
        }

        return $this;
    }

    public function removeReservation(Reservation $reservation): static
    {
        if ($this->reservation->removeElement($reservation)) {
            // set the owning side to null (unless already changed)
            if ($reservation->getDestination() === $this) {
                $reservation->setDestination(null);
            }
        }

        return $this;
    }
}
